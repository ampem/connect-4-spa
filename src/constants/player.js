export const RED_PLAYER = 'RED_PLAYER';
export const YELLOW_PLAYER = 'YELLOW_PLAYER';