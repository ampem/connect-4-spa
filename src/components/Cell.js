import React from 'react';
import PropTypes from 'prop-types';
import { RED_PLAYER } from '../constants/player';
import redCircle from '../assets/red.svg';
import yellowCircle from '../assets/yellow.svg';
import '../styles/cell.css';


function Cell({occupiedBy}) {
	const checkerImage = (occupiedBy === RED_PLAYER)?redCircle:yellowCircle;
    return (
        <div
          className="checker"
        >
        	<img src={checkerImage} className="checker-image" alt="checker" />
        </div>
       );
}

Cell.propTypes = {
	occupiedBy: PropTypes.string.isRequired
}

export default Cell;
