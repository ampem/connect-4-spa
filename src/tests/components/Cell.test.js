import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { RED_PLAYER, YELLOW_PLAYER } from '../../constants/player';
import Cell from '../../components/Cell';

describe('Cell', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<Cell occupiedBy={RED_PLAYER} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('Renders the yellow checker when occupiedBy is YELLOW_PLAYER', () => {
		const wrapper = shallow(<Cell occupiedBy={YELLOW_PLAYER} />);
		expect(wrapper).toMatchSnapshot();
	});
})
