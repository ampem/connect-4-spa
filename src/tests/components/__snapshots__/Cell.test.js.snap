// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell Renders the yellow checker when occupiedBy is YELLOW_PLAYER 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    occupiedBy="YELLOW_PLAYER"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="checker"
        className="checker-image"
        src="yellow.svg"
/>,
      "className": "checker",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "checker",
        "className": "checker-image",
        "src": "yellow.svg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="checker"
          className="checker-image"
          src="yellow.svg"
/>,
        "className": "checker",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "checker",
          "className": "checker-image",
          "src": "yellow.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Cell renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    occupiedBy="RED_PLAYER"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="checker"
        className="checker-image"
        src="red.svg"
/>,
      "className": "checker",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "checker",
        "className": "checker-image",
        "src": "red.svg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="checker"
          className="checker-image"
          src="red.svg"
/>,
        "className": "checker",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "checker",
          "className": "checker-image",
          "src": "red.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
